vcd/datasource_vcd_catalog_item.go

```diff
@@ -8,7 +8,8 @@ import (
 
 func datasourceVcdCatalogItem() *schema.Resource {
 	return &schema.Resource{
-		ReadContext: dataSourceVcdCatalogItemRead,
+		DeprecationMessage: "Use vcd_catalog_vapp_template instead",
```

Comment:
Marking as deprecated to suggest current consumers of this data source/resource that current implementation is moved to `vcd_catalog_vapp_template`. This way we can adjust the implementation of this one in next major.

Reply:
As @dataclouder mentions [here](https://github.com/vmware/terraform-provider-vcd/pull/899#discussion_r978601213), it is best to postpone these deprecation suggestions for the moment we enable the VM data source and resource to support this new vApp Template elements, hence I've removed the deprecation messages.